zebra-state/src/service/check/anchors.rs

```diff
@@ -3,193 +3,390 @@
 
 use std::{collections::HashMap, sync::Arc};
 
+use rayon::prelude::*;
+
 use zebra_chain::{
     block::{Block, Height},
-    sprout, transaction,
+    sprout,
+    transaction::{Hash as TransactionHash, Transaction, UnminedTx},
 };
 
 use crate::{
     service::{finalized_state::ZebraDb, non_finalized_state::Chain},
     PreparedBlock, ValidateContextError,
 };
 
-/// Checks the final Sapling and Orchard anchors specified by transactions in this
-/// `prepared` block.
+/// Checks the final Sapling and Orchard anchors specified by `transaction`
 ///
 /// This method checks for anchors computed from the final treestate of each block in
 /// the `parent_chain` or `finalized_state`.
-#[tracing::instrument(skip(finalized_state, parent_chain, prepared))]
-pub(crate) fn sapling_orchard_anchors_refer_to_final_treestates(
+#[tracing::instrument(skip_all)]
```

Comment:
The hash, index, and height are useful for debugging

```suggestion
#[tracing::instrument(skip(finalized_state, parent_chain, transaction))]
```